### PR TITLE
Add functionality to bind custom IP address for Etcd metrics endpoint

### PR DESCRIPTION
**Pull request by yuriydzobak**
#### Proposed Changes ####
k3s doesn't allow users to scrape metrics for etcd because it's bonded to localhost 

#### Types of Changes ####

Added to cli flag to change bind address for etcd metrics
#### Verification ####

```
k3s server --etcd-expose-metrics true
```

### CODE REVIEW

**Review by brandond — CHANGES_REQUESTED** [review]
Thanks for the contribution! We'd like to take a slightly different approach to it though. Rather than specifying a bind address in a new flag, can you instead add a new boolean flag (maybe `etcd-expose-metrics`) that when set will bind the metrics listener to the same address used for the peer and client listeners instead of loopback? Something like:

```golang
// metricsURL returns the metrics address for the local node
func (e *ETCD) clientURL(expose bool) string {
	if expose {
		return fmt.Sprintf("http://%s:2381", e.address)
	}
	return "http://127.0.0.1:2381"
}
```

File: pkg/cli/cmds/server.go [hunk 1]
```diff
@@ -209,6 +210,11 @@ func NewServerCommand(action func(*cli.Context) error) cli.Command {
 				Destination: &ServerConfig.DatastoreKeyFile,
 				EnvVar:      version.ProgramUpper + "_DATASTORE_KEYFILE",
 			},
+			&cli.BoolFlag{
+				Name:        "etcd-expose-metrics",
+				Usage:       "(dc) Expose etcd metrics to client interface. (Default false)",
```
Comment (brandond) [inline pkg/cli/cmds/server.go:215]:
```suggestion
				Usage:       "(db) Expose etcd metrics to client interface. (Default false)",
```

**Review by brandond — CHANGES_REQUESTED** [review]
One typo in the CLI flag, other than that looks great!

**Comment by yuriydzobak** [comment]
Hi, @brandond 
thanks for reviwe
Do you know why test was failed?
```

[SERIAL-mysql] Removing test directory /tmp/5xd8wX | 1322s
-- | --
3841 | [SERIAL-mysql] + '[' -f /tmp/tmp.peQzYg5FeG ']' | 1322s
3842 | [SERIAL-mysql] + echo | 1322s
3843 | [SERIAL-mysql] | 1322s
3844 | [SERIAL-mysql] ++ basename /tmp/5xd8wX | 1322s
3845 | [SERIAL-mysql] + echo -n 'Test 5xd8wX ' | 1322s
3846 | [SERIAL-mysql] + '[' 1 -eq 0 ']' | 1322s
3847 | [SERIAL-mysql] + echo failed. | 1322s
3848 | [SERIAL-mysql] + echo | 1322s
3849 | [SERIAL-mysql] Test 5xd8wX failed.
```

Maybe need to run re-test again

**Comment by brandond** [comment]
The sonobuoy tests frequently fail on flakey tests, I can restart it.

**Review by brandond — APPROVED** [review]

**Comment by yuriydzobak** [comment]
> The sonobuoy tests frequently fail on flakey tests, I can restart it.


Right now one test is failed =(
```
[PARALLEL-postgres] | 946s
-- | --
3616 | [PARALLEL-postgres] + '[' -f /tmp/tmp.aKXsVNp17g ']' | 946s
3617 | [PARALLEL-postgres] + echo | 946s
3618 | [PARALLEL-postgres] ++ basename /tmp/Pc7Dnt | 946s
3619 | [PARALLEL-postgres] + echo -n 'Test Pc7Dnt ' | 946s
3620 | [PARALLEL-postgres] + '[' 1 -eq 0 ']' | 946s
3621 | [PARALLEL-postgres] + echo failed. | 946s
3622 | [PARALLEL-postgres] Test Pc7Dnt failed.
```
Please, could you re-run the test again?
Thank you

**Review by briandowns — APPROVED** [review]

**Review by galal-hussein — APPROVED** [review]
LGTM

**Review by dweomer — APPROVED** [review]

**Pull request merged**